frontend/src/main/web/lib/components/SystemGlossary.jsx

```diff
@@ -0,0 +1,149 @@
+import React, { PureRenderMixin } from 'react/addons';
+import Configs from '../constants/Configs';
+import GlossaryStore from '../stores/GlossaryStore';
+import Actions from '../actions/GlossaryActions';
+import { Button, Input, Icons, Icon, Loader, Select } from 'zanata-ui'
+import DataTable from './glossary/DataTable'
+import NewEntryModal from './glossary/NewEntryModal'
+import ImportModal from './glossary/ImportModal'
+import _ from 'lodash';
+import StringUtils from '../utils/StringUtils'
+
+/**
+ * Main view for glossary page
+ */
+var SystemGlossary = React.createClass({
+  mixins: [PureRenderMixin],
+
+  filterTimeout: null,
+
+  _init: function() {
+    return GlossaryStore.init();
+  },
+
+  getInitialState: function() {
+    return this._init();
+  },
+
+  componentDidMount: function() {
+    GlossaryStore.addChangeListener(this._onChange);
+  },
+
+  componentWillUnmount: function() {
+    GlossaryStore.removeChangeListener(this._onChange);
+  },
+
+  _onChange: function() {
+    this.setState(this._init());
+  },
+
+  _onTranslationLocaleChange: function(localeId) {
+    this.setState({loadingEntries: true});
+    Actions.changeTransLocale(localeId)
+  },
+
+  _handleFilterKeyDown: function(event) {
+    if(event.key === 'Enter') {
+      Actions.updateFilter(this.state.filter);
+    }
+  },
+
+  _handleFilterChange: function(event) {
+    this.setState({filter: event.target.value});
+
+    if(this.filterTimeout !== null) {
+      clearTimeout(this.filterTimeout);
```

Comment:
You also need to clear this timeout when the component is going to unmount (in `componentWillUnmount`)